### ISSUE TITLE
[Serious] Keyboard focus is lost or misplaced due to user interaction or content update

### ISSUE THREAD

**Issue by ntallman** [APTrust/registry#104]
## Issue Details

| | |
|---|---|
| **Source Issue ID** | 87583 |
| **Checkpoint** | 2.4 Navigable › Focus Order (2.4.3.a) |
| **Impact** | Serious |
| **Standards** | wcag2a, wcag2aa, wcag21a, wcag21aa, wcag22a, wcag22aa |
| **Test Area** | 05 Add New User Workflow |
| **Test Case Name** | AP Trust Registry (Demo) |
| **URL** | <https://demo.aptrust.org/users> |
| **Environment** | Production |
| **Platform** | Windows | Chrome | Firefox |
| **Assistive Technology** | JAWS | NVDA | Magnifier |
| **Method** | Manual |
| **Digital Asset Type** | Desktop Web |
| **Release** | 4.2026 |
| **Product** | Website |
| **Date Reported** | 05/06/26 at 12:13 pm |
| **Unit Type** | Page |
| **Issue Type** | Accessibility |
| **Reported By** | a3c.access@gmail.com |

## Description

Keyboard focus is lost or placed on the wrong element during user interaction, content refresh or update, or other reason.
Pattern: Sorting a table column results in the keyboard focus jumping to the top of the page. For example, we selected the Email column header, but there was no confirmation that it has been sorted and the users focus jumps to the top of the page. See Managing Users User Experience Video at 1:55.
Also occurs on 03 Objects and 04 Work Items.

## Recommendation

Rule : 
The focus MUST be purposely moved or set (via JavaScript) onto the appropriate element when the user's action requires a change of context or location for effective keyboard or touch interaction.

How To Fix: 
Fix this issue by explicitly placing focus on a logical element when content is removed, refreshed, or added, for example:
1. For content added to the screen in reaction to a user-fired event, focus should be shifted to the new content - such as in single page applications.
2. For content removed from the screen in reaction to a user-fired event, focus should be shifted to the next logical place in the interaction.

Background: 
One of the biggest challenges when creating rich web interfaces using JavaScript is the management of focus when new content or controls are added or removed from the page. The presentation or modification of content on the screen may require that the user interact with -or at the very least, take notice of - the new or changed content. It is important to have a clear indication of page content that has been updated. This allows the user to understand the change and also allows a keyboard user to interact with any new content.

## Source Code

```
<th>
          <a href="/users?sort=email__desc" class="is-grey-dark">Email<span class="material-icons sort-icon" aria-hidden="true">
              keyboard_arrow_up</span></a>
        </th>
```

## Screenshots

![05_table sort](https://raw.githubusercontent.com/APTrust/registry/master/docs/accessibility-audit/screenshots/87583/05_table_sort.png)
